### PR TITLE
Update workflows from upstream and add missing ones

### LINKED ISSUE #606: Please update github action workflows
At least [update-nextcloud-ocp.yml](https://github.com/nextcloud/.github/blob/master/workflow-templates/update-nextcloud-ocp.yml) needs to be updated so you are informed about failures. My recent PR purged your composer.lock to allow installing the latest version, but you will miss when this breaks again, unless the action template is updated to the latest

### PR DESCRIPTION
**Pull request by come-nc**
Fix #606

### CODE REVIEW

File: .github/workflows/phpunit-mysql.yml [hunk 1]
```diff
@@ -94,7 +85,7 @@ jobs:
         with:
           php-version: ${{ matrix.php-versions }}
           # https://docs.nextcloud.com/server/stable/admin_manual/installation/source_installation.html#prerequisites-for-manual-installation
-          extensions: bz2, ctype, curl, dom, fileinfo, gd, iconv, intl, json, libxml, mbstring, openssl, pcntl, posix, session, simplexml, xmlreader, xmlwriter, zip, zlib, mysql, pdo_mysql, ldap
+          extensions: bz2, ctype, curl, dom, fileinfo, gd, iconv, intl, json, libxml, mbstring, openssl, pcntl, posix, session, simplexml, xmlreader, xmlwriter, zip, zlib, mysql, pdo_mysql
```
Comment (nickvergessen) [inline .github/workflows/phpunit-mysql.yml:88]:
```suggestion
          extensions: bz2, ctype, curl, dom, fileinfo, gd, iconv, intl, json, libxml, mbstring, openssl, pcntl, posix, session, simplexml, xmlreader, xmlwriter, zip, zlib, mysql, pdo_mysql, ldap
```

File: .github/workflows/phpunit-mysql.yml [hunk 2]
```diff
@@ -123,7 +114,6 @@ jobs:
         run: |
           mkdir data
           ./occ maintenance:install --verbose --database=mysql --database-name=nextcloud --database-host=127.0.0.1 --database-port=$DB_PORT --database-user=root --database-pass=rootpassword --admin-user admin --admin-pass admin
-          ./occ app:enable --force user_ldap
```
Comment (nickvergessen) [inline .github/workflows/phpunit-mysql.yml:126]: revert

File: .github/workflows/phpunit-mysql.yml [hunk 3]
```diff
@@ -155,8 +145,6 @@ jobs:
         # Only run if phpunit integration config file exists
         if: steps.check_integration.outcome == 'success'
         working-directory: apps/${{ env.APP_NAME }}
-        env:
-          LDAP_HOST: localhost
```
Comment (nickvergessen) [inline .github/workflows/phpunit-mysql.yml:159]: revert

**Review by nickvergessen — CHANGES_REQUESTED** [review]

File: .github/workflows/phpunit-mysql.yml (hunk 2 above)
Comment (come-nc) [reply .github/workflows/phpunit-mysql.yml:126]: Thank you, I got confused in my file management and forgot to revert this one :+1:

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**